tests/test_unit/test_cdf_tk/test_resource_classes/test_robotics/test_capabilities.py

```diff
@@ -0,0 +1,70 @@
+from collections.abc import Iterable
+from pathlib import Path
+
+import pytest
+
+from cognite_toolkit._cdf_tk.resource_classes.robotics.capabilities import RobotCapabilityYAML
+from cognite_toolkit._cdf_tk.tk_warnings.fileread import ResourceFormatWarning
+from cognite_toolkit._cdf_tk.validation import validate_resource_yaml_pydantic
+from tests.test_unit.utils import find_resources
+
+
+def valid_robot_capability_cases() -> Iterable:
+    yield pytest.param(
+        {
+            "name": "PTZ Camera",
+            "externalId": "ptz_camera_001",
+            "method": "capture_ptz_image",
+            "description": "Pan-Tilt-Zoom camera capability for capturing images",
+            "inputSchema": {
+                "type": "object",
+                "properties": {
+                    "pan": {"type": "number", "minimum": -180, "maximum": 180},
+                    "tilt": {"type": "number", "minimum": -90, "maximum": 90},
+                    "zoom": {"type": "number", "minimum": 1, "maximum": 10},
+                },
+            },
+            "dataHandlingSchema": {
+                "uploadLocation": "cdf_files",
+                "fileFormat": "jpeg",
+                "metadata": {"captureType": "ptz"},
+            },
+        },
+        id="full-ptz-camera-capability",
+    )
+    yield pytest.param(
+        {"name": "Docking Station", "externalId": "dock_001", "method": "return_to_dock"}, id="minimal-robot-capability"
+    )
+
+
+def invalid_robot_capability_cases() -> Iterable:
+    yield pytest.param(
+        {"name": "Missing External ID", "method": "some_method"},
+        {"Missing required field: 'externalId'"},
+        id="missing-required-field",
+    )
+    yield pytest.param(
+        {"externalId": "cap_001", "name": "Test Capability", "method": "test_method", "unknownField": "value"},
+        {"Unused field: 'unknownField'"},
+        id="unknown-field-present",
+    )
+
+
+class TestRobotCapabilityYAML:
+    @pytest.mark.parametrize("data", list(find_resources("RobotCapability")))
+    def test_load_valid_robot_capability_file(self, data: dict[str, object]) -> None:
+        loaded = RobotCapabilityYAML.model_validate(data)
+        assert loaded.model_dump(exclude_unset=True, by_alias=True) == data
+
+    @pytest.mark.parametrize("data", valid_robot_capability_cases())
+    def test_load_valid_robot_capability(self, data: dict[str, object]) -> None:
+        loaded = RobotCapabilityYAML.model_validate(data)
+        assert loaded.model_dump(exclude_unset=True, by_alias=True) == data
+
+    @pytest.mark.parametrize("data, expected_errors", invalid_robot_capability_cases())
```

Comment:
I don't see the value of having these fixtures (e.g. `invalid_robot_capability_cases`), since the developer need to scroll the page to see what paramethers are. why not passing them as a list of  `pytest.param` directly to the `pytest.mark.parametrize`